auth/idtoken/idtoken.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package idtoken
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"cloud.google.com/go/auth"
+	"cloud.google.com/go/auth/internal"
+	"cloud.google.com/go/auth/internal/internaldetect"
+	"cloud.google.com/go/compute/metadata"
+)
+
+// ComputeTokenFormat dictates the the token format when requesting an ID token
+// from the compute metadata service.
+type ComputeTokenFormat int
+
+const (
+	// ComputeTokenFormatDefault means the same as [ComputeTokenFormatFull].
+	ComputeTokenFormatDefault ComputeTokenFormat = iota
+	// ComputeTokenFormatStandard mean only standard JWT fields will be included
+	// in the token.
+	ComputeTokenFormatStandard
+	// ComputeTokenFormatFull means the token will include claims about the
+	// virtual machine instance and its project.
+	ComputeTokenFormatFull
+	// ComputeTokenFormatFullWithLicense means the same as
+	// [ComputeTokenFormatFull] with the addition of claims about licenses
+	// associated with the instance.
+	ComputeTokenFormatFullWithLicense
+)
+
+// Options for the configuration of creation of an ID token with
+// [NewTokenProvider].
+type Options struct {
```

Comment:
I love the new dedicated `Options` struct as opposed to the old design which used shared client Options.

Reply:
:)